staker/staker.go

```diff
@@ -291,9 +291,10 @@ func (s *Staker) Start(ctxIn context.Context) {
 		arbTx, err := s.Act(ctx)
 		if err == nil && arbTx != nil {
 			_, err = s.l1Reader.WaitForTxApproval(ctx, arbTx)
-			err = fmt.Errorf("error waiting for tx receipt: %w", err)
 			if err == nil {
 				log.Info("successfully executed staker transaction", "hash", arbTx.Hash())
+			} else {
+				err = fmt.Errorf("error waiting for tx receipt: %w", err)
```

Comment:
I also fixed another issue I noticed in testing, where the github.com/pkg/errors -> standard library errors change broke this code, as errors.Wrap will return nil if the error is nil but fmt.Errorf always returns a non-nil error